csp_billing_adapter_microsoft/plugin.py

```diff
@@ -32,16 +32,12 @@
 
 log = logging.getLogger('CSPBillingAdapter')
 
-METADATA_ADDR = 'http://169.254.169.254/metadata'
-METADATA_API_VERSION = '?api-version=2021-12-13'
-METADATA_INSTANCE_PATH = '/instance'
-METADATA_TOKEN_PATH = '/identity/oauth2/token'
-METADATA_TOKEN_RESOURCE = '&resource=https://management.azure.com/'
-
-METADATA_INSTANCE_URL = METADATA_ADDR + METADATA_INSTANCE_PATH
-METADATA_TOKEN_URL = \
-    METADATA_ADDR + METADATA_TOKEN_PATH + METADATA_API_VERSION + \
-    METADATA_TOKEN_RESOURCE
+METADATA_URL = 'http://169.254.169.254/metadata/'
+COMPUTE_URL = METADATA_URL + 'instance/compute?api-version=2017-08-01'
```

Comment:
Why are we using this api-version "2017-08-01" ? The metadata API offers a route to get the list of supported values for  api-version.  Should we at least query the list and pick the earliest (2017-03-01) or latest (2021-12-13) as opposed to some version in the middle (2017-08-01)

curl -s -H Metadata:true --noproxy "*" "http://169.254.169.254/metadata/versions"
{"apiVersions":["2017-03-01","2017-04-02","2017-08-01","2017-10-01","2017-12-01","2018-02-01","2018-04-02","2018-10-01","2019-02-01","2019-03-11","2019-04-30","2019-06-01","2019-06-04","2019-08-01","2019-08-15","2019-11-01","2020-06-01","2020-07-15","2020-09-01","2020-10-01","2020-12-01","2021-01-01","2021-02-01","2021-03-01","2021-05-01","2021-08-01","2021-10-01","2021-11-01","2021-11-15","2021-12-13"]}

Also I need the api-version for when I request an access_token so wouldn't it be better to put this in its own variable to avoid duplication?

Reply:
I will update it, API version should be `latest`